packages/components/src/toolbar/toolbar-item/index.tsx

```diff
@@ -15,8 +14,18 @@ import warning from '@wordpress/warning';
  * Internal dependencies
  */
 import ToolbarContext from '../toolbar-context';
+import type { ToolbarItemProps } from './types';
+import type { WordPressComponentProps } from '../../ui/context';
 
-function ToolbarItem( { children, as: Component, ...props }, ref ) {
+function ToolbarItem(
+	{
+		children,
+		as,
+		...props
+	}: WordPressComponentProps< ToolbarItemProps, 'div', false >,
+	ref: ForwardedRef< any >
+) {
+	const Component = as;
```

Comment:
Could easily clean this code up using `<As />`, which I've seen in another project, instead of recasting it to `Component`.